src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -334,13 +334,29 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
                         _logger.LogWarning("App is configured to use both Run-From-Package and AzureFiles. Run-From-Package will take precedence");
                     }
                     var blobContextApplied =
-                        await _runFromPackageHandler.ApplyBlobPackageContext(pkgContext, options.ScriptPath,
+                        await _runFromPackageHandler.ApplyRunFromPackageContext(pkgContext, options.ScriptPath,
                             azureFilesMounted, false);
 
                     if (!blobContextApplied && azureFilesMounted)
                     {
-                        _logger.LogWarning($"Failed to {nameof(_runFromPackageHandler.ApplyBlobPackageContext)}. Attempting to use local disk instead");
-                        await _runFromPackageHandler.ApplyBlobPackageContext(pkgContext, options.ScriptPath, false);
+                        _logger.LogWarning($"Failed to {nameof(_runFromPackageHandler.ApplyRunFromPackageContext)}. Attempting to use local disk instead");
+                        await _runFromPackageHandler.ApplyRunFromPackageContext(pkgContext, options.ScriptPath, false);
+                    }
+                }
+                else if (pkgContext.IsRunFromLocalPackage())
+                {
+                    if (!azureFilesMounted)
+                    {
+                        _logger.LogWarning("App Run-From-Package is set as '1'. AzureFiles is needed but is not configured.");
```

Comment:
if RFP=1, and no AF is configured, shouldnt we fail right away? where are we copying the zip from in this case?

Reply:
I am now failing it if FileShare is not mounted.